### PR TITLE
Add local event log storage

### PR DESCRIPTION
**Pull request by ellie**
Start storing an event log locally! 

This will allow us to eventually start syncing events, rather than pure history items.

When merging an event log, we will pair up CREATEs with DELETEs. If there is a pair, we won't create the history. If there is not a pair (and just a CREATE), then the item is created. I've also started on the server end of this, while I had the context in my head. Obviously it's no use just yet

I've been running this branch locally for a bit, and have had no issues.

### CODE REVIEW

File: atuin-client/src/database.rs [hunk 1]
```diff
@@ -115,6 +118,27 @@ impl Sqlite {
         Ok(())
     }
 
+    async fn save_event(tx: &mut sqlx::Transaction<'_, sqlx::Sqlite>, e: &Event) -> Result<()> {
+        let event_type = match e.event_type {
+            EventType::Create => "create",
+            EventType::Delete => "create",
```
Comment (conradludgate) [inline atuin-client/src/database.rs:124]: Typo

File: atuin-client/src/database.rs [hunk 2]
```diff
@@ -302,6 +330,49 @@ impl Database for Sqlite {
         Ok(res)
     }
 
+    async fn event_count(&self) -> Result<i64> {
+        let res: (i64,) = sqlx::query_as("select count(1) from events")
```
Comment (conradludgate) [inline atuin-client/src/database.rs:334]: I believe this works
```suggestion
        let res: i64 = sqlx::query_scalar("select count(1) from events")
```

File: atuin-client/src/database.rs [hunk 3]
```diff
@@ -302,6 +330,49 @@ impl Database for Sqlite {
...
+            // But also even if you have a LOT of history that should be fine, and we're only going to be doing this once EVER.
+            let all_the_history = self
+                .list(FilterMode::Global, &no_context, None, false)
+                .await?;
+
+            let mut tx = self.pool.begin().await?;
+            for i in all_the_history.iter() {
+                // A CREATE for every single history item is to be expected.
+                let event = Event::new_create(i);
+                Self::save_event(&mut tx, &event).await?;
```
Comment (conradludgate) [inline atuin-client/src/database.rs:368]: From what I understand, you're relying on sqlite to say ignore the errors when inserting duplicate events? From what I know on Postgres, transactions are dead once there is an error

Comment (conradludgate) [reply atuin-client/src/database.rs:368]: Never mind, the save event method ignores the failures explicitly

Comment (ellie) [reply atuin-client/src/database.rs:368]: Indeed. Way easier to let the unique index and SQLite handle deduping

File: atuin-client/src/database.rs (hunk 2 above)
Comment (ellie) [reply atuin-client/src/database.rs]: That's super neat! Thanks

**Review by conradludgate — APPROVED** [review]

**Pull request merged**